### PR TITLE
Create Zendesk tickets when user requests to merge companies

### PR DESCRIPTION
**Pull request by rafenden**
## Description of change

Create Zendesk tickets when users request to merge two companies. Before they had to send a support request via web form.

## Test instructions

1. Navigate to an unmatched company
2. Click "Verify business details"
3. Select a company that already exists on DH
4. Click "Request merge" on the next page
5. A Zendesk ticket should be created on our Zendesk staging env

### Before

image

### After

image

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/lib/urls.js [hunk 1]
```diff
@@ -137,6 +137,8 @@ module.exports = {
     match: {
       index: url('/companies', '/:companyId/match'),
       confirmation: url('/companies', '/:companyId/match/:dunsNumber'),
+      link: url('/companies', '/:companyId/match/link'),
+      merge: url('/companies', '/:companyId/match/merge'),
```
Comment (debitan) [inline src/lib/urls.js]: think you just need to add a quick couple of tests for these.

Comment (rafenden) [reply src/lib/urls.js]: This route is covered in `match-company-spec.js`.

File: src/apps/companies/apps/match-company/client/MatchDuplicate.jsx [hunk 2]
```diff
@@ -1,50 +1,74 @@
...
+            />
+          )}
+
+          <p>
+            This can happen when there are duplicate company records in Data
+            Hub. To resolve this, you can ask the Support Team to merge these
+            duplicates into one record.
+          </p>
+
+          <H4 as="h2">Requesting records merged will:</H4>
```
Comment (web-bert) [inline src/apps/companies/apps/match-company/client/MatchDuplicate.jsx]: I assume this should be merge and not merged?

File: src/apps/companies/apps/match-company/controllers.js [hunk 3]
```diff
@@ -208,10 +212,56 @@ async function renderCannotFindMatch(req, res, next) {
...
+
+    req.flash(
+      'success',
+      'Company merge requested. Thanks for keeping Data Hub running smoothly.'
+    )
+    res.json({ message: 'OK', ticket: get(result, 'data.ticket.id') })
+  } catch (error) {
+    const statusCode = get(error, 'response.status', 500)
+    const message = get(error, 'response.data.description')
+    res.status(statusCode).json({ message })
```
Comment (web-bert) [inline src/apps/companies/apps/match-company/controllers.js]: Would it be a good idea to log this error to Sentry using `src/lib/reporter.js`?

File: src/lib/urls.js (hunk 1 above)
Comment (web-bert) [reply src/lib/urls.js]: I think in `src/lib/__test_/urls.test.js` is where some tests need to be?

Comment (rafenden) [reply src/lib/urls.js]: I'm not sure if that's the case, @mcbhenwood and @peterhudec decided a while ago to stop writing unit tests if the functionality is covered by functional tests.

File: src/apps/companies/apps/match-company/client/MatchDuplicate.jsx (hunk 2 above)
Comment (rafenden) [reply src/apps/companies/apps/match-company/client/MatchDuplicate.jsx]: Good point, I updated the content.

File: src/apps/companies/apps/match-company/controllers.js (hunk 3 above)
Comment (rafenden) [reply src/apps/companies/apps/match-company/controllers.js]: Usually, we are not using sentry for this kind of errors.

Comment (web-bert) [reply src/apps/companies/apps/match-company/controllers.js]: So how do you know if this fails or you don't care?

Comment (rafenden) [reply src/apps/companies/apps/match-company/controllers.js]: That's a good point, I updated the code to `next(error)`.

This way the sentry middleware will report this exception.

**Pull request merged**